sql/api/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBaseParser.g4

```diff
@@ -643,7 +643,7 @@ sortItem
     ;
 
 fromStatement
-    : fromClause fromStatementBody+
+    : fromClause (fromStatementBody)*
```

Comment:
I need a bit more context to understand this change. The classic SQL does not require additional clauses after FROM, e.g. `... FROM t` is valid. Is this parser rule only used by pipe syntax?

Reply:
@cloud-fan The parser rule is used by both pipe SQL syntax as well as regular SQL syntax. Currently, we support the `fromStatement` is a valid entire SQL query:

```
queryPrimary
    : querySpecification         #queryPrimaryDefault
    | fromStatement              #fromStmt
    ...

fromStatement
    : fromClause fromStatementBody+
    ;

fromStatementBody
    : transformClause
      whereClause?
      queryOrganization
    | selectClause
      lateralView*
      whereClause?
      ...
    ;
```

So for example, `FROM t SELECT a, b` is a valid Spark SQL query.

Therefore it is also valid to write `FROM t SELECT * |> WHERE a = 42`.

This PR is making the second part optional, so you could write `FROM t` as equivalent to `TABLE t` to return the entire table, or `FROM t |> WHERE a = 42`.

Let us know if this is OK to you or if we should restrict the grammar changes to only applying to SQL pipe syntax only.